src/librustc/ty/mod.rs

```diff
@@ -2824,18 +2824,29 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         self.trait_def(trait_def_id).has_auto_impl
     }
 
+    /// Returns the trait corresponding to the given trait alias.
+    pub fn get_trait_from_alias(self, def_id: DefId) -> Option<ty::TraitDef> {
```

Comment:
What is the purpose of this?

Reply:
Err... I think I was just tinkering around. Ignore this. :-)

Reply:
Actually, it was almost right. I need a fn for checking if a def-id is for a trait alias.